src/components/PressableWithSecondaryInteraction/index.js

```diff
@@ -1,103 +1,124 @@
 import _ from 'underscore';
-import React, {Component} from 'react';
-import * as pressableWithSecondaryInteractionPropTypes from './pressableWithSecondaryInteractionPropTypes';
+import React, {forwardRef, useEffect, useRef} from 'react';
 import styles from '../../styles/styles';
 import * as DeviceCapabilities from '../../libs/DeviceCapabilities';
 import * as StyleUtils from '../../styles/StyleUtils';
 import PressableWithFeedback from '../Pressable/PressableWithFeedback';
+import * as pressableWithSecondaryInteractionPropTypes from './pressableWithSecondaryInteractionPropTypes';
 
 /**
  * This is a special Pressable that calls onSecondaryInteraction when LongPressed, or right-clicked.
  */
-class PressableWithSecondaryInteraction extends Component {
-    constructor(props) {
-        super(props);
-        this.executeSecondaryInteraction = this.executeSecondaryInteraction.bind(this);
-        this.executeSecondaryInteractionOnContextMenu = this.executeSecondaryInteractionOnContextMenu.bind(this);
-    }
-
-    componentDidMount() {
-        if (this.props.forwardedRef) {
-            if (_.isFunction(this.props.forwardedRef)) {
-                this.props.forwardedRef(this.pressableRef);
-            } else if (_.isObject(this.props.forwardedRef)) {
-                this.props.forwardedRef.current = this.pressableRef;
-            }
-        }
-        this.pressableRef.addEventListener('contextmenu', this.executeSecondaryInteractionOnContextMenu);
-    }
 
-    componentWillUnmount() {
-        this.pressableRef.removeEventListener('contextmenu', this.executeSecondaryInteractionOnContextMenu);
-    }
+function PressableWithSecondaryInteraction({
+    children,
+    inline,
+    style,
+    enableLongPressWithHover,
+    withoutFocusOnSecondaryInteraction,
+    preventDefaultContextMenu,
+    onSecondaryInteraction,
+    onPressIn,
+    onPress,
+    onPressOut,
+    activeOpacity,
+    forwardedRef,
+    ...rest
+}) {
+    const pressableRef = useRef(null);
 
     /**
      * @param {Event} e - the secondary interaction event
      */
-    executeSecondaryInteraction(e) {
-        if (DeviceCapabilities.hasHoverSupport() && !this.props.enableLongPressWithHover) {
+    const executeSecondaryInteraction = (e) => {
+        if (DeviceCapabilities.hasHoverSupport() && !enableLongPressWithHover) {
             return;
         }
-        if (this.props.withoutFocusOnSecondaryInteraction && this.pressableRef) {
-            this.pressableRef.blur();
+        if (withoutFocusOnSecondaryInteraction && pressableRef && pressableRef.current) {
```

Comment:
Nitpick: I think that was redundant and we can simplify like this

```suggestion
        if (withoutFocusOnSecondaryInteraction && pressableRef.current) {
```

Reply:
won't it throw an error if `pressableRef` is null?